### PR TITLE
Refactor mandatory-claims to enforced-claims

### PR DESCRIPTION
**Pull request by tzheleznyak**
Refactor mandatory-claims to enforced-claims

### What ticket does this PR close?
Resolves #[relevant GitHub issues, eg 76]

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: design/authenticators/authn_jwt/token_schema.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Solution Design - Authn JWT Mandatory Claims And Mapping
+# Solution Design - Authn JWT enforced Claims And Mapping
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:1]:
```suggestion
# Solution Design - Authn JWT Enforced Claims And Mapping
```

File: design/authenticators/authn_jwt/token_schema.md [hunk 2]
```diff
@@ -8,7 +8,7 @@
 ## Table of Contents
 [//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
 
-- * [Solution Design - Authn JWT Mandatory Claims And Mapping](#solution-design---authn-jwt-mandatory-claims-and-mapping)
+- * [Solution Design - Authn JWT enforced Claims And Mapping](#solution-design---authn-jwt-enforced-claims-and-mapping)
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:11]:
```suggestion
- * [Solution Design - Authn JWT Enforced Claims And Mapping](#solution-design---authn-jwt-enforced-claims-and-mapping)
```

File: design/authenticators/authn_jwt/token_schema.md [hunk 3]
```diff
@@ -120,11 +120,11 @@ When validate_restrictions is called in JWTVendorConfiguration class the followi
 
 1. The `validate_restrictions` function will call `CreateConstraintsFromPolicy` that create constraints object checking the hosts annotations.
    1. It create instance of `NonEmptyConstraint` to check we have at least one host annotation.
-   2. Will call `LoadMandatoryClaims` command class to load the list of mandatory claims for the `mandatory-claims` variable. This class will create and return `RequiredConstraint` object.
+   2. Will call `LoadenforcedClaims` command class to load the list of enforced claims for the `enforced-claims` variable. This class will create and return `RequiredConstraint` object.
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:123]:
```suggestion
   2. Will call `LoadMandatoryClaims` command class to load the list of enforced claims for the `enforced-claims` variable. This class will create and return `RequiredConstraint` object.
```
We don't change non user faced names. Class name remain the same, ins't it?

File: design/authenticators/authn_jwt/token_schema.md [hunk 4]
```diff
@@ -147,7 +147,7 @@ When validate_restrictions is called in JWTVendorConfiguration class the followi
 ## Backwards Compatibility
 [//]: # "How will the design of this solution impact backwards compatibility? Address how you are going to handle backwards compatibility, if necessary"
 
-Mandatory claims and mapping are not mandatory therefore the JWT authentication will remain backwards compatible.
+enforced claims and mapping are not enforced therefore the JWT authentication will remain backwards compatible.
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:150]:
```suggestion
Enforced claims and mapping are not enforced therefore the JWT authentication will remain backwards compatible.
```

File: design/authenticators/authn_jwt/token_schema.md [hunk 5]
```diff
@@ -163,20 +163,20 @@ Mandatory claims and mapping are not mandatory therefore the JWT authentication
 
 | Mission                                                      | Estimation |
 | ------------------------------------------------------------ | ---------- |
-| Add LoadMandatoryClaims                                      | 1 SP       |
+| Add LoadenforcedClaims                                      | 1 SP       |
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:166]: The same. Class name remains the same

**Review by sashaCher — COMMENTED** [review]
Leave couple comments.
Next time please user case-sensitive find-replace.

File: design/authenticators/authn_jwt/token_schema.md (hunk 5 above)
Comment (tzheleznyak) [reply design/authenticators/authn_jwt/token_schema.md:166]: I will change class name

File: design/authenticators/authn_jwt/token_schema.md [hunk 6]
```diff
@@ -120,11 +120,11 @@ When validate_restrictions is called in JWTVendorConfiguration class the followi
 
 1. The `validate_restrictions` function will call `CreateConstraintsFromPolicy` that create constraints object checking the hosts annotations.
    1. It create instance of `NonEmptyConstraint` to check we have at least one host annotation.
-   2. Will call `LoadMandatoryClaims` command class to load the list of mandatory claims for the `mandatory-claims` variable. This class will create and return `RequiredConstraint` object.
+   2. Will call `LoadeEforcedClaims` command class to load the list of enforced claims for the `enforced-claims` variable. This class will create and return `RequiredConstraint` object.
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:123]:
```suggestion
   2. Will call `LoadEforcedClaims` command class to load the list of enforced claims for the `enforced-claims` variable. This class will create and return `RequiredConstraint` object.
```

File: app/domain/authentication/authn_jwt/consts.rb [hunk 7]
```diff
@@ -28,10 +28,10 @@ module AuthnJwt
     CACHE_REFRESHES_PER_INTERVAL = 10
     CACHE_RATE_LIMIT_INTERVAL = 300
     CACHE_MAX_CONCURRENT_REQUESTS = 3
-    MANDATORY_CLAIMS = [EXP_CLAIM_NAME].freeze
+    ENFORCED_CLAIMS = [EXP_CLAIM_NAME].freeze
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/consts.rb:31]: No it's really mandatory claims :-)

File: app/domain/authentication/authn_jwt/consts.rb [hunk 8]
```diff
@@ -28,10 +28,10 @@ module AuthnJwt
...
     OPTIONAL_CLAIMS = [ISS_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME].freeze
     VALID_CLAIM_NAME_REGEX = /^[a-zA-Z|$|_][a-zA-Z|$|_|0-9]*$/.freeze # starts with letter $ or _ can contains digits
-    MANDATORY_CLAIMS_DENY_LIST = [ISS_CLAIM_NAME, EXP_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME, JTI_CLAIM_NAME, AUD_CLAIM_NAME].freeze
+    ENFORCED_CLAIMS_DENY_LIST = [ISS_CLAIM_NAME, EXP_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME, JTI_CLAIM_NAME, AUD_CLAIM_NAME].freeze
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/consts.rb:34]:
```suggestion
    CLAIMS_DENY_LIST = [ISS_CLAIM_NAME, EXP_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME, JTI_CLAIM_NAME, AUD_CLAIM_NAME].freeze
```
Mapping is also using the value

File: app/domain/authentication/authn_jwt/input_validation/parse_mandatory_claims.rb [hunk 9]
```diff
@@ -1,82 +1,82 @@
 module Authentication
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/input_validation/parse_mandatory_claims.rb:1]: File name ^

File: design/authenticators/authn_jwt/authn_jwt_solution_design.md [hunk 10]
```diff
@@ -195,7 +195,7 @@ The `FactoryFetchSignigKey` class will be responsible on the following logic:
 ###### Fetch JWT standard claims to validate
 For 1st milestone we will validate the following:
 1. nbf iat - will be validated if exists in token, (not extra value is  needed from customer)
-2. exp - it is mandatory, if not exists in the token we will throw an error
+2. exp - it is enforced, if not exists in the token we will throw an error
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/authn_jwt_solution_design.md:198]: exp is a real mandatory claim

File: design/authenticators/authn_jwt/authn_jwt_solution_design.md [hunk 11]
```diff
@@ -118,7 +118,7 @@ implementation no matter what are the answers of the following questions:
    identity (user/host) policy, but according to this design we have the
    flexibility to support other approaches in the future.
    
-    4.2. What are the constraints of the restrictions? mandatory? optional? permitted ? at least one? OR relation? AND relation? 
+    4.2. What are the constraints of the restrictions? enforced? optional? permitted ? at least one? OR relation? AND relation?
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/authn_jwt_solution_design.md:121]: Let's keep an initial design as is.

File: app/domain/authentication/authn_jwt/validate_and_decode/fetch_jwt_claims_to_validate.rb [hunk 12]
```diff
@@ -5,7 +5,7 @@ module AuthnJwt
     module ValidateAndDecode
       # FetchJwtClaimsToValidate command class is responsible to return a list of JWT standard claims to
       # validate, according to the following logic:
-      # For each optional claim (iss, exp, nbf, iat) that exists in the token - add to mandatory list
+      # For each optional claim (iss, exp, nbf, iat) that exists in the token - add to enforced list
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/validate_and_decode/fetch_jwt_claims_to_validate.rb:8]: Roll back the file please.
It deals with mandatory claims.

File: design/authenticators/authn_jwt/token_schema.md [hunk 13]
```diff
@@ -1,36 +1,13 @@
...
 [//]: # "1. Design should be graphical-based and table-based - avoid long text explanations"
 [//]: # "2. Design documents should not be updated after implementation"
-[//]: # "3. Design decisions should be made before writing this document, and as such this document should not include options / choices"
+[//]: # "3.  Design decisions should be made before writing this document, and as such this document should not include options / choices"
 
 
 ## Table of Contents
 [//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
 
-- * [Solution Design - Authn JWT Mandatory Claims And Mapping](#solution-design---authn-jwt-mandatory-claims-and-mapping)
```
Comment (sashaCher) [inline design/authenticators/authn_jwt/token_schema.md:11]: Table of contents

File: spec/app/domain/authentication/authn-jwt/validate_and_decode/fetch_jwt_claims_to_validate_spec.rb [hunk 14]
```diff
@@ -85,7 +85,7 @@ def token(claims)
   #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
 
   context "JWT decoded token input" do
-    context "with mandatory claims (exp)" do
+    context "with enforced claims (exp)" do
```
Comment (sashaCher) [inline spec/app/domain/authentication/authn-jwt/validate_and_decode/fetch_jwt_claims_to_validate_spec.rb:88]: The same there the class deals with mandatory claims like `exp`

**Review by sashaCher — APPROVED** [review]
LGTM

**Pull request merged**